### PR TITLE
Stop suggesting adding generic args for turbofish

### LINKED ISSUE #100137: Incorrect suggestion when type annotation on turbofished enum is needed
Given the following code: https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=b9b341939839ec733bf081c5ddf86e5b

```rust
enum OhNo<T, U> {
  A(T),
  B(U),
  C,
}

fn uwu() {
    OhNo::C::<u32, _>;
}
```

The current output is:

```rust
Compiling playground v0.0.1 (/playground)
error[E0282]: type annotations needed
 --> src/lib.rs:8:5
  |
8 |     OhNo::C::<u32, _>;
  |     ^^^^^^^^^^^^^^^^^ cannot infer type of the type parameter `U` declared on the enum `OhNo`
  |
help: consider specifying the generic arguments
  |
8 |     OhNo::<u32, _>::C::<u32, _>;
  |         ++++++++++

For more information about this error, try `rustc --explain E0282`.
error: could not compile `playground` due to previous error
```

But applying this causes an error, saying that `C` doesn't allow type arguments.

This happens on nightly and beta, but not on stable. I'm assuming the "consider specifying the generic arguments" was added recently.

I don't think we need the help here, because the generic arguments are already specified.

@rustbot label +D-invalid-suggestion

### PR DESCRIPTION
**Pull request by TaKO8Ki**
Fixes #100137

### CODE REVIEW

File: src/test/ui/inference/need_type_info/do-not-suggest-generic-arguments-for-turbofish.stderr [hunk 1]
```diff
@@ -0,0 +1,9 @@
+error[E0282]: type annotations needed
+  --> $DIR/do-not-suggest-generic-arguments-for-turbofish.rs:8:5
+   |
+LL |     OhNo::C::<u32, _>;
+   |     ^^^^^^^^^^^^^^^^^ cannot infer type of the type parameter `U` declared on the enum `OhNo`
```
Comment (TaKO8Ki) [inline src/test/ui/inference/need_type_info/do-not-suggest-generic-arguments-for-turbofish.stderr:5]: I think this suggestion should point at the specific type parameter. I will work on it later.

**Comment by lcnr** [comment]
I personally like the suggestions, even if the generic args are already specified, but include `_`. Maybe with a slightly changed wording. If you get an error with `collect::<Vec<_>>()` the correct fix would be `collect::<Vec<MyType>>()`, so we should still point at this `collect` call instead of searching for something else.

A more targeted fix for #100137 would be to not add an `InsertableGenericArgs` for enum types if there are explicit generic arguments on the variant. Because `OhNo::<u32, _>::C` is absolutely fine, and even preferred over `OhNo::C::<u32, _>`. This only breaks if the generic arguments are specified for both the enum and the variant

File: compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs [hunk 2]
```diff
@@ -998,7 +1009,14 @@ impl<'a, 'tcx> FindInferSourceVisitor<'a, 'tcx> {
                     }
                     let span = tcx.hir().span(segment.hir_id?);
                     let insert_span = segment.ident.span.shrink_to_hi().with_hi(span.hi());
-                    InsertableGenericArgs { insert_span, substs, generics_def_id: def_id, def_id }
+                    debug!(?insert_span, "fhaowehfoaehfoahweo");
```
Comment (compiler-errors) [inline compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs:1012]: can you give this a real message :stuck_out_tongue_winking_eye:

**Comment by compiler-errors** [comment]
r=me with nit fixed

**Pull request merged**